File: typescript/src/headers/exceptions/invalid-header-exception.ts
/**
 * Thrown when a serialized header cannot be parsed.
 *
 * @public
 * @exported
 * @extends {Error}
 * @param {number} maxEntityId - The maximum entityId value that can be serialized.
 */
export class InvalidHeaderException extends Error {
    constructor(maxEntityId: number) {
        super(`The entityId must be an integer between 0 and ${ maxEntityId } because the header serializer uses exactly four bits for entityId value.`);

        this.name = new.target.name;
    }
}

File: dotnet/Barchart.BinarySerializer/Headers/Exceptions/InvalidHeaderException.cs
namespace Barchart.BinarySerializer.Headers.Exceptions;

/// <summary>
///     Thrown when a serialized header cannot be parsed.
/// </summary>
public class InvalidHeaderException : InvalidOperationException
{
    #region Constructor(s)
    
    /// <summary>
    ///     Initializes a new instance of the <see cref="InvalidHeaderException"/> class.
    /// </summary>
    /// <param name="maxEntityId">
    ///     The maximum entityId value that can be used.
    /// </param>
    public InvalidHeaderException(byte maxEntityId) : base($"The entityId cannot exceed {maxEntityId} because the header serializer uses exactly four bits for entityId value.")
    {
        
    }
    
    #endregion
}